experimental/bits/simd_builtin.h

```diff
@@ -2837,7 +2840,7 @@ template <typename _Abi>
 
     // smart_reference access {{{2
     template <typename _Tp, size_t _Np>
-      static constexpr void _S_set(_SimdWrapper<_Tp, _Np>& __k, int __i,
+      static constexpr void _S_set(_SimdWrapper<_Tp, _Np>& __k, size_t __i,
```

Comment:
All `_S_set` functions declare the index parameter as `int`. To resolve the warning please change `__i == __j` to `__i == int(__j)`. In any case, the warning is a false positive, since `__j` is a constant expression thus the situation Clang warns about cannot happen.

Reply:
I think about that, but the index will be expected to be a negative value?
 if not why not replace "int" with size_t?

Reply:
True, `_M_set` using `size_t` while the rest uses `int` is inconsistent. Wrt. choosing the type: I could equally ask whether the index is expected to be a value greater than `INT_MAX`. Neither that, nor negative values are ever going to happen. The better question to ask: does this type model a mathematical integer where a + 1 > a or should it model a modulo ring where a + 1 typically is greater than a but sometimes may also be less than a. We don't need the latter. The former allows the compiler to reason better about the type and thus lead to better optimizations.
This is why the default integer type to use should always be signed. That the C++ standard library uses `size_t` for most indexes and size parameters is widely recognized as a mistake that we cannot fix nowadays.